python/ray/dashboard/modules/job/job_agent.py

```diff
@@ -118,14 +121,31 @@ async def delete_job(self, req: Request) -> Response:
                 text=f"Job {job_or_submission_id} does not exist",
                 status=aiohttp.web.HTTPNotFound.status_code,
             )
-        if job.type is not JobType.SUBMISSION:
-            return Response(
-                text="Can only delete submission type jobs",
-                status=aiohttp.web.HTTPBadRequest.status_code,
-            )
-
         try:
-            deleted = await self.get_job_manager().delete_job(job.submission_id)
+            job_mgr = self.get_job_manager()
+            if job.type is JobType.DRIVER:
+                deleted = await job_mgr.delete_from_job_table(job.job_id)
+            elif job.type is JobType.SUBMISSION:
```

Comment:
the implementation seems pretty inefficient, is it possible to just find a job id from `delete_from_job_table ` and directly delete it from GCS using await job_mgr.delete_from_job_table(job_item.job_id.hex())?

Reply:
Indeed been an issue, has been optimized.